src/frontend/src/optimizer/plan_visitor/relation_collector_visitor.rs

```diff
@@ -52,22 +52,16 @@ impl PlanVisitor for RelationCollectorVisitor {
     }
 
     fn visit_batch_seq_scan(&mut self, plan: &crate::optimizer::plan_node::BatchSeqScan) {
-        if !plan.core().is_sys_table() {
-            self.relations.insert(plan.core().table_desc.table_id);
-        }
+        self.relations.insert(plan.core().table_desc.table_id);
     }
 
     fn visit_logical_scan(&mut self, plan: &LogicalScan) {
-        if !plan.is_sys_table() {
-            self.relations.insert(plan.table_desc().table_id);
-        }
+        self.relations.insert(plan.table_desc().table_id);
```

Comment:
I don't get this logic previously 😆 why insert the same thing twice if it's a system table cc @yezizp2012

Reply:
Okay nevermind, I confused myself.

Reply:
I read it as:
```
        if !plan.is_sys_table() {
            self.relations.insert(plan.table_desc().table_id);
        }
        self.relations.insert(plan.table_desc().table_id);
```
But it's a diff....